aspnetcore/performance/caching/middleware.md

```diff
@@ -1,15 +1,114 @@
+Pragma | <p>The response is not cached if the value is set to `no-cache`. This header is overridden by the relevant directives in the `Cache-Control` header if present.</p><p>Considered for backward compatibility with HTTP/1.0.</p>
+Set-Cookie | <p>The response is not cached if the header exists.</p>
+Vary | <p>You can vary the cached response by another header. For example, you can cache responses by encoding by including the `Vary: Accept-Encoding` header, which would cache responses to requests with headers `Accept-Encoding: gzip` and `Accept-Encoding: text/plain` separately. A response with a header value of `*` is never stored.</p>
+Expires | <p>A response deemed stale by this header will not be stored or retrieved unless overridden by other `Cache-Control` headers.</p>
+If-None-Match | <p>The full response will be served from cache if the value is not `*` and the `ETag` of the response doesn't match any of the values provided. Otherwise, a 304 (Not Modified) response will be served.</p>
+If-Modified-Since | <p>If the `If-None-Match` header is not present, a full response will be served from cache if the cached response date is newer than the value provided. Otherwise, a 304 (Not Modified) response will be served.</p>
+Date | <p>When serving from cache, the `Date` header is set by the middleware if it wasn't provided on the original response.</p>
+Content-Length | <p>When serving from cache, the `Content-Length` header is set by the middleware if it wasn't provided on the original response.</p>
+Age | <p>The `Age` header sent in the original response will be ignored. The middleware will compute a new value when serving a cached response.</p>
+
+## Troubleshooting
+If caching behavior is not as you expect, confirm that responses are cacheable and capable of being served from the cache by examining the request's incoming headers and the response's outgoing headers. The conditions by which a response will be cached or served from the cache are listed below.
+
+When testing and troubleshooting caching behavior, a browser may set request headers that affect caching in undesirable ways. For example, a browser may set the `Cache-Control` header to `no-cache` when you refresh the page. Instead of using a browser, use a tool like [Fiddler](http://www.telerik.com/fiddler), [Firebug](http://getfirebug.com/), or [Postman](https://www.getpostman.com/), all of which allow you to explicitly set request headers.
+
+* The request must result in a 200 (OK) response from the server.
+* The request method must be GET or HEAD.
+* Terminal middlewares, such as Static File Middleware, must not process the response prior to the Response Caching Middleware.
+* The Authorization header must not be present.
+* `Cache-Control` header parameters must be valid, and the response must be marked `public`.
```

Comment:
It should be obvious but let's also mention that we don't cache responses marked as `private` for completeness here. The caching middleware is operates as a "public" cache.